tests/api_connexion/endpoints/test_dag_run_endpoint.py

```diff
@@ -1421,7 +1421,7 @@ def test_should_respond_200(self, dag_maker, session):
             "execution_date": dr.execution_date.isoformat(),
             "external_trigger": False,
             "logical_date": dr.logical_date.isoformat(),
-            "start_date": dr.logical_date.isoformat(),
```

Comment:
This feels like it shouldn't have changed.

Reply:
Okay this is right due to this code, and it shouldn't have haver been passing before! (it was missing a flush or a proper refresh before.)

https://github.com/apache/airflow/blob/fb7c6afc8cb7f93909bd2e654ea185eb6abcc1ea/airflow/models/taskinstance.py#L276-L278

Reply:
https://github.com/apache/airflow/pull/27849/files/68696c2f7edf1e9727b4b4c348245c138e05038f#diff-97c63a6aeab204e63d50b7a4d44807c3019257ad272925d1e7074c01879a7c8aL415

Reply:
but i didn't want to load the notes in all usages of `refresh_from_db`.  so better to not use the helper.